src/activity-types/enigme.constants.ts

```diff
@@ -7,146 +7,156 @@ export const ENIGME = {
-      label: 'Un instrument de musique',
-      description: 'Faites nous découvrir un instrument de musique.',
-      step: 'Un instrument de musique',
-    },
-    {
-      label: 'Un costume',
-      description: 'Faites nous découvrir un costume mystère.',
-      step: 'Un costume',
-    },
-  ],
-  [
-    {
-      label: 'Une fête de l’école',
-      description: 'Faites nous découvrir les jours de fêtes de votre école.',
-      step: 'Une fête',
-    },
-    {
-      label: 'Un festival',
-      description: 'Et si vous tentiez de nous faire découvrir un festival typique de chez vous ?',
-      step: 'Un festival',
-    },
-    {
-      label: 'Un fait historique',
-      description: 'Présentez ici un fait historique s’étant déroulé dans votre région.',
-      step: 'Un fait historique',
-    },
-    {
-      label: 'Une fête nationale',
-      description: 'Faites nous découvrir une fête de votre pays.',
-      step: 'Une fête nationale',
-    },
-  ],
-  [
-    {
-      label: 'Un personnage historique',
-      description: 'Faites nous découvrir un personnage historique.',
-      step: 'Un personnage historique',
-    },
-    {
-      label: 'Un personnage de fiction',
-      description: 'Faites nous découvrir un personnage de fiction.',
-      step: 'Un personnage de fiction',
-    },
-    {
-      label: 'Un personnage contemporain',
-      description: 'Faites nous découvrir un personnage contemporain.',
-      step: 'Un personnage contemporain',
-    },
-  ],
-  [
-    {
-      label: 'Un monument commémoratif',
-      description: 'Faites nous découvrir un monument commémoratif.',
-      step: 'Un monument commémoratif',
-    },
-    {
-      label: 'Un édifice remarquable',
-      description: 'Faites nous découvrir un édifice remarquable.',
-      step: 'Un édifice remarquable',
-    },
-    {
-      label: 'Une salle de spectacle',
-      description: 'Faites nous découvrir une salle de spectacle.',
-      step: 'Une salle de spectacle',
-    },
-    {
-      label: "Un ouvrage d'art",
-      description: "Faites nous découvrir un ouvrage d'art.",
-      step: "Un ouvrage d'art",
-    },
-  ],
-];
-
 export function getEnigmeTimeLeft(activity: EnigmeActivity): number {
   return 7 - Math.floor((new Date().getTime() - new Date(activity.updateDate || '').getTime()) / 3600000 / 24);
 }
 
-export const getEnigme = (activitySubType: number | null | undefined, activityData: EnigmeData) =>
-  (activitySubType === -1 ? { title: activityData.themeName || '', step1: activityData.themeName || '' } : ENIGME_TYPES[activitySubType || 0]) || {
-    title: 'Enigme',
-    step1: 'Enigme',
-  };
+// This function is used only in step 1
+export function getCategoryName(activitySubType: number | null | undefined, activityData: EnigmeData) {
+  return activitySubType === -1 // this is when we choose a category 'autre' in index page
+    ? { title: activityData.themeName || 'Thème' }
+    : { title: ENIGME_TYPES[activitySubType || 0].title };
+}
```

Comment:
The logic is great! But there's no need to return an object with `title`, you can directly return the string:

**Suggestion:**
```typescript
export function getCategoryName(activitySubType: number | null | undefined, activityData: EnigmeData): string {
  return activitySubType === -1 // this is when we choose a category 'autre' in index page
    ? activityData.themeName || 'Thème'
    : ENIGME_TYPES[activitySubType || 0].title
}
```